apps/build-a-pinterest-clone/Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM node:16-bullseye-slim
+
+WORKDIR /app
+
+# Copy over all the files in the project directory to /app early
+# for rollup bundling
+COPY . . 
+
+ENV PORT=3000
+ENV GITHUB_KEY=${BUILD_A_PINTEREST_CLONE_GITHUB_KEY}
+ENV GITHUB_SECRET=${BUILD_A_PINTEREST_CLONE_GITHUB_SECRET}
+ENV DB_URI=${BUILD_A_PINTEREST_CLONE_DB_URI}
+ENV APP_URL=${BUILD_A_PINTEREST_CLONE_APP_URL}
```

Comment:
Why do we repeat all the env variables in each Dockerfile and `.env` file?

Reply:
That's to make the "container deployment" platform agnostic.

1. Docker Compose in a single VM - Use a `.env` file and startup the apps.
2. Azure App Service / AWS Fargate or something similar - just set the respective values in the UI and point it to the registry and the image.

It looks a bit ugly code-wise, but that's worth the flexibility.